generator/osm2type.cpp

```diff
@@ -319,15 +319,34 @@ class CachedTypes
 // e.g. highway-primary-bridge is left while highway-primary is removed.
 void LeaveLongestTypes(vector<generator::TypeStrings> & matchedTypes)
 {
-  auto const less = [](auto const & lhs, auto const & rhs) { return lhs > rhs; };
+  auto const equalPrefix = [](auto const & lhs, auto const & rhs)
+  {
+    size_t const prefixSz = std::min(lhs.size(), rhs.size());
+    bool const res = equal(lhs.begin(), lhs.begin() + std::min(size_t(2), prefixSz), rhs.begin());
+    if (res && prefixSz > 2)
+    {
+      // Define the set of exceptions like this:
+      if (lhs[1] == "charging_station")
```

Comment:
why `charging_station` needs an exception but other types of size>2 don't?

Reply:
Our classifier assumes unique x-y-* types rather than combined x-y-1 + x-y-2 in most cases. If there are more exceptions, let me know.